src/main/pg/pg.c

```diff
@@ -49,7 +49,7 @@ void pgResetInstance(const pgRegistry_t *reg, uint8_t *base)
     const uint16_t regSize = pgSize(reg);
 
     memset(base, 0, regSize);
-    if (reg->reset.ptr >= (void*)__pg_resetdata_start && reg->reset.ptr < (void*)__pg_resetdata_end) {
+    if (reg->reset.ptr >= (void*)&__pg_resetdata_start && reg->reset.ptr < (void*)&__pg_resetdata_end) {
```

Comment:
not sure here. maybe '&start[0]'? Declaring start as something with size feels wrong

Reply:
Result:

```bash
compiling ../main/pg/pg.c 
../main/pg/pg.c:52:27: error: cast to 'void *' from smaller integer type 'uint8_t' (aka 'unsigned char') [-Werror,-Wint-to-void-pointer-cast]
    if (reg->reset.ptr >= (void*)__pg_resetdata_start && reg->reset.ptr < (void*)__pg_resetdata_end) {
                          ^~~~~~~~~~~~~~~~~~~~~~~~~~~
../main/pg/pg.c:52:75: error: cast to 'void *' from smaller integer type 'uint8_t' (aka 'unsigned char') [-Werror,-Wint-to-void-pointer-cast]
    if (reg->reset.ptr >= (void*)__pg_resetdata_start && reg->reset.ptr < (void*)__pg_resetdata_end) {
                                                                          ^~~~~~~~~~~~~~~~~~~~~~~~~
```

Reply:
declaration as array[] (as before this pr) is necessary.

Reply:
@ledvinap fixed!